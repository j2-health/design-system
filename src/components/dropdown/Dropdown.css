.ant-dropdown .ant-dropdown-menu .ant-dropdown-menu-item {
  color: var(--j2-color-primary-text);

  &:hover {
    color: var(--j2-color-primary-text-hover);
  }
}
