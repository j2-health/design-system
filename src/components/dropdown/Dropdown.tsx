import { Dropdown as AntdDropdown, DropdownProps, Typography } from 'antd'
import { Button } from '../button'
import { CaretDownIcon } from '@phosphor-icons/react'
import './Dropdown.css'

const { Link } = Typography

export type Props = DropdownProps & {
  type?: 'basic' | 'basic-inline' | 'twofold'
  label?: string
  icon?: React.ReactNode
}

const Label = ({ label }: { label: string | undefined }) => (
  <div className="flex items-center gap-2">
    {label}
    <CaretDownIcon weight="regular" />
  </div>
)

const Target = ({ label, type, disabled, icon }: Props) => {
  if (type === 'basic-inline') {
    return (
      <Link disabled={disabled}>
        {icon ? (
          <>
            {icon}
            {label}
          </>
        ) : (
          <Label label={label} />
        )}
      </Link>
    )
  }

  return (
    <Button disabled={disabled} icon={icon}>
      {icon ? label : <Label label={label} />}
    </Button>
  )
}

const Dropdown = ({ label, type, icon, ...props }: Props) => {
  if (type === 'twofold') {
    return (
      <AntdDropdown.Button {...props}>
        {icon ? (
          <>
            {icon} {label}
          </>
        ) : (
          label
        )}
      </AntdDropdown.Button>
    )
  }

  return (
    <AntdDropdown {...props}>
      <div>
        <Target
          label={label}
          type={type}
          disabled={props.disabled}
          icon={icon}
        />
      </div>
    </AntdDropdown>
  )
}

export { Dropdown }
