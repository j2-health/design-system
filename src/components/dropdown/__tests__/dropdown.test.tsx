import { render } from '@testing-library/react'
import { Dropdown } from '../Dropdown'

describe('Dropdown', () => {
  it('should render correctly', () => {
    const { container } = render(<Dropdown label="Test">Test</Dropdown>)
    expect(container).toMatchSnapshot()
  })
})
