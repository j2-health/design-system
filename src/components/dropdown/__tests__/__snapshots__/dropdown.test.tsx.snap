// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render correctly 1`] = `
<div>
  <div
    class="ant-dropdown-trigger"
  >
    <button
      class="ant-btn css-dev-only-do-not-override-ccdg5a ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
      type="button"
    >
      <span>
        Test
      </span>
    </button>
  </div>
</div>
`;
